Fastlane/deployment_lanes.rb

```diff
@@ -0,0 +1,473 @@
+desc " - `DEVELOPER_KEY_PATH`: The path to the file containing the private key"
+desc ""
+lane :authenticate do | options |
+	use_app_manager_role = options[:use_app_manager_role] || false
+	key_id = use_app_manager_role ? ENV["APP_MANAGER_KEY_ID"] : ENV["DEVELOPER_KEY_ID"]
+	key_filepath = use_app_manager_role ? ENV["APP_MANAGER_KEY_PATH"] : ENV["DEVELOPER_KEY_PATH"]
+	issuer_id = ENV["JWT_ISSUER_ID"]
+
+	UI.important "Authenticating using #{use_app_manager_role ? "App Manager Role" : "Developer Role"}"
+
+  app_store_connect_api_key(
+    key_id: key_id,
+    issuer_id: issuer_id,
+    key_filepath: key_filepath,
+    duration: 1200, # 90 minutes, matching Bitrise timeout limit.
+    in_house: false, # optional but may be required if using match/sigh
+  )
+end
+
+
+desc "Returns true if there are new changes since the last available tag"
+private_lane :is_changed_since_last_tag do
+  sh "git fetch --tags origin #{ENV["BITRISE_GIT_BRANCH"]} --no-recurse-submodules"
+  last_tag = last_git_tag
+  changes = sh "git diff --name-only HEAD #{last_tag}"
+  puts "Is local HEAD changed since last tag #{last_tag}: #{!changes.empty?}"
+  is_changed = !changes.empty?
+end
+
+desc "Updates the build number of the project based on the commit count"
+desc ""
+desc "#### Options"
+desc " * **`xcodeproj`**: the path to the Xcode project to read the version number from"
+desc " * **`target`**: The target within the project to read the version number from"
+desc ""
+private_lane :update_build_number do |options|
+	# Get the path to the Xcode project and the target to read the version number from.
+	xcodeproj = options[:xcodeproj]
+	target = options[:target]
+
+  # Compute a new build number based on the commit count.
+	build = sh "git rev-list --all --count"
+	new_build_number = 200 + Integer(build)
+
+	# Fetch the build number of the most recent test flight build, to make sure
+	# that the new build number will be higher.
+	version_number = get_version_number(xcodeproj: xcodeproj, target: target)
+	latest_app_store_build_number = latest_testflight_build_number
+	test_flight_build_number = latest_testflight_build_number(version: version_number)
+
+  if new_build_number <= latest_app_store_build_number
+    puts "git build number is smaller than the build number of the latest release"
+    puts "Using the latest release build number #{latest_app_store_build_number} + 1"
+    new_build_number = latest_app_store_build_number + 1
+  end
+
+  if new_build_number <= test_flight_build_number
+    puts "git build number is smaller than test flight build number"
+    puts "Using the TestFlight build number #{test_flight_build_number} + 1"
+    new_build_number = test_flight_build_number + 1
+  end
+
+  increment_build_number(build_number: new_build_number)
+end
+
+desc "Syncs production certificates on CI"
+desc ""
+desc "#### Options"
+desc " * **`app_identifier`**: a list of all app identifiers for which to sync the certs"
+desc ""
+private_lane :certs do |options|
```

Comment:
Shouldn't this be part of the provisioning lanes file?

Reply:
I decided to make it a private lane as it should only be used on CI. The provisioning lanes can be used locally on the command line. Let me know if that makes sense.

Reply:
wasn't aware of private lanes, awesome!